subprojects/docs/src/snippets/tasks/incrementalBuild-customTaskClass/groovy/build.gradle

```diff
@@ -23,12 +27,18 @@ tasks.register('processTemplatesAdHoc') {
     outputs.dir(layout.buildDirectory.dir('genOutput2'))
         .withPropertyName('outputDir')
 
+// end::ad-hoc-task[]
```

Comment:
Added a few tags for hiding details that are not related to the documentation section.